pkg/po/lex.go

```diff
@@ -187,26 +216,78 @@ func parseToken(line string, tokenType TokenType) (*Token, error) {
 
-	return &token, nil
+	return token, nil
 }
 
-// parseValue parses the value of the token.
+// parseTokenValue extracts and parses the value of a token.
 // e.g.
 //
+//	line -> value
 //	"Language: en-US\n" -> en-US\n
 //	msgid "some text" -> some text
 //	#, python-format -> python-format
 //	#: superset/key_value/exceptions.py:54 -> superset/key_value/exceptions.py:54
-func parseValue(line string) string {
-	n := 2
-	fields := strings.SplitN(line, " ", n) // prefix and value, e.g. fields[0] = msgid, fields[1] = "text", etc.
+//
+// Returns the parsed token value and a nil error if successful, or an error if parsing fails.
+func parseTokenValue(linePrefix, line string, tokenType TokenType) (string, error) {
+	value := strings.TrimPrefix(line, linePrefix)
+
+	switch tt := tokenType; tt {
+	case TokenTypeMsgID, TokenTypePluralMsgID, TokenTypeMsgStr, TokenTypeMsgCtxt, TokenTypePluralMsgStr:
+		if !strings.HasPrefix(value, " ") {
+			return "", fmt.Errorf("token type '%d': value must be prefixed with space", tt)
+		}
+
+		value = strings.TrimSpace(value)
+
+		// value must be enclosed in double quotes
+		if !(strings.HasPrefix(value, "\"") && strings.HasSuffix(value, "\"")) {
+			return "", fmt.Errorf("token type '%d': value must be enclosed in double quotation mark - \"\" ", tt)
+		}
+
+		value = strings.TrimPrefix(strings.TrimSuffix(value, "\""), "\"")
+	case TokenTypeHeaderLanguage, TokenTypeHeaderTranslator, TokenTypeHeaderPluralForms,
+		TokenTypeHeaderProjectIDVersion, TokenTypeHeaderPOTCreationDate, TokenTypeHeaderPORevisionDate,
+		TokenTypeHeaderLanguageTeam, TokenTypeHeaderLastTranslator, TokenTypeHeaderXGenerator,
+		TokenTypeHeaderReportMsgidBugsTo, TokenTypeHeaderMIMEVersion, TokenTypeHeaderContentType,
+		TokenTypeHeaderContentTransferEncoding, TokenTypeHeaderGeneratedBy:
+		value = strings.TrimSpace(value)
+
+		if !strings.HasSuffix(value, "\"") {
+			return "", fmt.Errorf("token type '%d': line must end with double quotation mark - \" ", tt)
+		}
+
+		value = strings.TrimSuffix(value, "\"")
+	case TokenTypeTranslatorComment, TokenTypeExtractedComment, TokenTypeReference, TokenTypeFlag:
+		if !(len(value) == 0 || strings.HasPrefix(value, " ")) {
+			return "", fmt.Errorf("token type '%d': value must be prefixed with space", tt)
+		}
+
+		value = strings.TrimSpace(value)
+	case TokenTypeMsgidPluralPrevUntStrPlural, TokenTypeMsgctxtPreviousContext, TokenTypeMsgidPrevUntStr:
+		if !(len(value) == 0 || strings.HasPrefix(value, " ")) {
+			return "", fmt.Errorf("token type '%d': value must be prefixed with space", tt)
+		}
+
+		value = strings.TrimSpace(value)
 
-	// No value, only prefix
-	if len(fields) != n {
-		return ""
+		if len(value) != 0 {
+			value = " " + value
+		}
+
+		switch tt { //nolint:exhaustive
+		case TokenTypeMsgidPluralPrevUntStrPlural:
+			value = "msgid_plural" + value
+		case TokenTypeMsgctxtPreviousContext:
+			value = "msgctxt" + value
+		case TokenTypeMsgidPrevUntStr:
+			value = "msgid" + value
```

Comment:
why this is here ? 🤔 
parseTokenValue extracts the value of the token, but here the value is also prefixed with tokenType.
1. I don't think it is correct
2. It contradicts with function godoc, particular with: `msgid "some text" -> some text`

Reply:
well in the happy path tests -> 		mkToken(TokenTypeMsgidPrevUntStr, "msgid previous id"),

Reply:
dont think that is correct :)